src/queryRates.ts

```diff
@@ -1,273 +1,54 @@
-import request from 'graphql-request';
-import { formatFixed } from '@ethersproject/bignumber';
-import { lnWad, WAD } from './FixedPointMathLib';
+import { Rate } from './types';
+import fetchMarketState from './api';
+import { apr, apy, marketStateUtilization } from './utils';
 
-const FIXED_INTERVAL = 86_400 * 7 * 4;
-const PRECISION_THRESHOLD = 750_000_000_000_000n;
-
-const min = (a: bigint, b: bigint) => (a < b ? a : b);
-const max = (a: bigint, b: bigint) => (a > b ? a : b);
-
-const floatingRate = (
-  interestRateModel: InterestRateModel,
-  utilizationBefore: bigint,
-  utilizationAfter: bigint,
-) => {
-  const curveA = BigInt(interestRateModel.floatingCurveA);
-  const curveB = BigInt(interestRateModel.floatingCurveB);
-  const maxUtilization = BigInt(interestRateModel.floatingMaxUtilization);
-  const alpha = maxUtilization - utilizationBefore;
-  const delta = utilizationAfter - utilizationBefore;
-
-  return ((delta * WAD) / alpha < PRECISION_THRESHOLD
-    ? ((curveA * WAD) / alpha
-      + (curveA * 4n * WAD) / (maxUtilization - ((utilizationAfter + utilizationBefore) / 2n))
-      + (curveA * WAD) / (maxUtilization - utilizationAfter)) / 6n
-    : (curveA * lnWad((alpha * WAD) / (maxUtilization - utilizationAfter))) / delta
-  ) + curveB;
-};
-
-const totalFloatingBorrowAssets = (
-  timestamp: number,
-  { floatingAssets, floatingDebt }: MarketState,
-  { timestamp: debtUpdate, utilization }: FloatingDebtState,
-  interestRateModel: InterestRateModel,
-) => {
-  const newUtilization = BigInt(floatingAssets) > 0n
-    ? (BigInt(floatingDebt) * WAD) / BigInt(floatingAssets)
-    : 0n;
-  const borrowRate = floatingRate(
-    interestRateModel,
-    min(BigInt(utilization), newUtilization),
-    max(BigInt(utilization), newUtilization),
+export const rate = async (
```

Comment:
no need to export it
```suggestion
const rate = async (
```